test/index/vars/vars.snapshot.cc

```diff
@@ -39,3 +39,22 @@
     return s.x + S::y;
 //         ^ reference local 8
   }
+  
+  void lambdas() {
+    int x = 0;
+//      ^ definition local 9
+    int y = 1;
+//      ^ definition local 10
+    auto add = [&x, y](int z) mutable {
```

Comment:
The one slightly iffy thing here is, we're not emitting a separate local definition here in the capture list. Since `y` is being copied, it's essentially creating a new local variable, and the `y` below is a reference to this `y`, not the `y` earlier.

@dominiccooney thoughts on we should keep the current behavior or change it?

Go to definition in VS Code by clangd (arguably incorrectly) takes one from the `y` reference in the body to the `y` outside the lambdas (and not the capture list), even when capturing-by-value. CLion also behaves similar to VS Code.